File: src/main/java/ru/reosfire/WMLib/Guis/Inventory/Components/DelegateButton.java
package ru.reosfire.WMLib.Guis.Inventory.Components;

import org.bukkit.event.inventory.InventoryClickEvent;
import ru.reosfire.WMLib.Guis.Inventory.Gui;
import ru.reosfire.WMLib.Guis.Inventory.IClickEventHandler;
import ru.reosfire.WMLib.Yaml.Default.Gui.ButtonConfig;

public class DelegateButton extends Button
{
    private final IClickEventHandler Delegate;
    public DelegateButton(ButtonConfig config, Gui gui, IClickEventHandler delegate)
    {
        super(config, gui);
        Delegate = delegate;
    }

    @Override
    protected void OnClick(InventoryClickEvent event)
    {
        Delegate.Handle(event);
    }
}

File: src/main/java/ru/reosfire/wmlib/guis/inventory/components/DelegateButton.kt
package ru.reosfire.wmlib.guis.inventory.components

import org.bukkit.event.inventory.InventoryClickEvent
import ru.reosfire.wmlib.yaml.common.gui.ButtonConfig
import ru.reosfire.wmlib.guis.inventory.Gui
import ru.reosfire.wmlib.guis.inventory.IClickEventHandler

class DelegateButton(config: ButtonConfig, gui: Gui, private val delegate: IClickEventHandler) : Button(config, gui) {
    override fun onClick(event: InventoryClickEvent) {
        delegate.handle(event)
    }
}